spikeinterface/sorters/basesorter.py

```diff
@@ -0,0 +1,270 @@
+        Main function keept for backward compatibility.
+        This should not be used anymore.
+        """
+        # setup recording
+        self._setup_recording(self.recording, self.output_folder)
+        
+        # compute
+        self.compute_from_folder(self.output_folder, raise_error=True)
+
+    @staticmethod
+    def compute_from_folder(output_folder, raise_error=True):
+        # need setup_recording to be done.
+        output_folder = Path(output_folder)
+        
+        # retrieve sorter and params
+        with (output_folder / 'spikeinterface_params.json').open(mode='r') as f:
+            params = json.load(f)
+        sorter_params = params['sorter_params']
+        sorter_name = params['sorter_name']
+        verbose = params['verbose']
+
+        from .sorterlist import sorter_dict
+        SorterClass = sorter_dict[sorter_name]
+        
+        # not needed normally
+        # recording = load_extractor(output_folder / 'spikeinterface_recording.json')
+
+        now = datetime.datetime.now()
+        log = {
+            'sorter_name': str(SorterClass.sorter_name),
+            'sorter_version': str(SorterClass.get_sorter_version()),
+            'datetime': now,
+            'runtime_trace': []
+        }
+        t0 = time.perf_counter()
+        
+        #~ SorterClass._compute_from_folder(output_folder, sorter_params, verbose)
+        
+        try:
+            SorterClass._compute_from_folder(output_folder, sorter_params, verbose)
+            t1 = time.perf_counter()
+            run_time = float(t1 - t0)
+            has_error = False
+        except Exception as err:
+            has_error = True
+            run_time = None
+            log['error'] = True
+            log['error_trace'] = traceback.format_exc()
+        
+        log['error'] = has_error
+        log['run_time'] = run_time
+
+        # some sorter have a log file dur to shellscript launcher
+        runtime_trace_path = output_folder / f'{sorter_name}.log'
+        runtime_trace = []
+        if runtime_trace_path.is_file():
+            with open(runtime_trace_path, 'r') as fp:
+                line = fp.readline()
+                while line:
+                    runtime_trace.append(line.strip())
+                    line = fp.readline()
+        log['runtime_trace'] = runtime_trace
+        
+        # dump to json
+        with (output_folder / 'spikeinterface_log.json').open('w', encoding='utf8') as f:
+            json.dump(check_json(log), f, indent=4)
+
+        if verbose:
+            if has_error:
+                print(f'Error running {sorter_name}')
+            else:
+                print(f'{sorter_name} run time {run_time:0.2f}s')
+
+        if has_error and raise_error:
+            raise SpikeSortingError(f"Spike sorting failed. You can inspect the runtime trace in "
+                                    f"the {sorter_name}.log of the output folder.'")
+        
+        return run_time
+
+    def get_result(self):
```

Comment:
note that the `raise_error` need to be propagated here too: see https://github.com/SpikeInterface/spikesorters/pull/207